### DISCUSSION TITLE
Shell command invocation

### DISCUSSION THREAD

**Discussion by lakemike** [bdd/runitor#13] [Q&A] [answered]
Dear Berk, all,

I am still trying to get runitor to execute a command string that includes a pipe.

I tried both:
> runitor -api-url=https://example.com/api -- my_cmd --flag=y foo | grep -v error:
and
> runitor -api-url=https://example.com/api -- bash -c "my_cmd --flag=y foo | grep -v error:"

Problem is that neither of them works i.e. runitor is not picking up stdout.

Any advice? Despite >30y bash experience I struggle with this one. But then .. it's bash .. and sometimes it just gives you a headache or you don't see the tree in the middle of the forest.

I'd appreciate if someone could share something that works in another install.

Thanks!

**Comment by bdd** [comment, marked as answer]
Runitor will copy both stdout and stderr to the ping body in v.0.6.0 unless commanded otherwise.

Can you tell me what version you're running? `runitor -version` would tell you. If you see `runitor HEAD`, it means you built a non-dist version yourself. It's fine but it would be easier for me to try to reproduce issues if you build from a tag, or even simpler use prebuilt binaries from releases https://github.com/bdd/runitor/releases/. 0.6.0 is the latest stable release.

Testing with 0.6.0:
```
% git switch -c v0.6.0
% ./scripts/build dist
% ./build/runitor-v0.6.0-darwin-amd64 -- bash -c "(echo out; echo foo; echo err >&2) | grep -v foo"
err
out
```

It worked as intended for the terminal output. We wanted to filter out "foo" from stdout.  "err" went to stderr, and "out" went to stdout.

If we check what it send to healthchecks.io instance:  
![image](https://user-images.githubusercontent.com/11135/106394918-2b7ff000-63b4-11eb-83f8-9019c733d18e.png)

So... It works as intended.

Can you paste your actual command line instead of these examples? Feel free to redact your instance's URL.

**Reply by lakemike** [reply]
Very helpful! I had followed the link to "latest release" on runitor front page and had started to use beta version 0.7.0. Indeed, it works as you describe above once I go back to stable v0.6. Huge thanks for the help. Shall I post this as bug for beta v0.7.0?

**Reply by bdd** [reply]
Please do with the exact reproduction steps. This sounds like a rather embarrassing bug that's not getting caught with existing unit tests.

**Reply by bdd** [reply]
BTW, I can't reproduce it with 0.7.0-beta.4.

![image](https://user-images.githubusercontent.com/11135/106397140-bbc43200-63c0-11eb-8f90-62c703c89eb7.png)

**Reply by lakemike** [reply]
I can reproduce it on my end. It's on a debian buster system with either runitor-v0.7.0-beta.4-linux-amd64 or runitor-v0.6.0-linux-amd64
